Build failure: sm64ex

### Nixpkgs version

- Unstable (25.05)

### Steps to reproduce

1. Build `sm64ex`
2. Get error

### Can Hydra reproduce this build failure?

No, Hydra cannot reproduce this build failure.

### Link to Hydra build job

_No response_

### Relevant log output

```console
error: builder for '/nix/store/i3b7x4yd0aw0vip390zzq8lf6b19h9i6-sm64ex-0-unstable-2024-07-04.drv' failed with exit code 2;
       last 50 log lines:
       > tools/n64graphics -i build/us_pc/levels/wf/0.rgba16 -g levels/wf/0.rgba16.png -f rgba16
       > tools/n64graphics -i build/us_pc/levels/wf/1.rgba16 -g levels/wf/1.rgba16.png -f rgba16
       > tools/n64graphics -i build/us_pc/levels/wf/2.rgba16 -g levels/wf/2.rgba16.png -f rgba16
       > tools/n64graphics -i build/us_pc/levels/wf/3.rgba16 -g levels/wf/3.rgba16.png -f rgba16
       > tools/n64graphics -i build/us_pc/levels/wf/4.rgba16 -g levels/wf/4.rgba16.png -f rgba16
       > In file included from src/pc/gfx/gfx_sdl2.c:22:
       > /nix/store/vjrb1ayf85l4b7xr21fi9kkdgxrk4fnc-sdl2-compat-2.32.52-dev/include/SDL2/SDL_opengles2.h:34:10: fatal error: GLES2/gl2platform.h: No such file or directory
       >    34 | #include <GLES2/gl2platform.h>
       >       |          ^~~~~~~~~~~~~~~~~~~~~
       > compilation terminated.
       > tools/n64graphics -i build/us_pc/levels/wf/5.ia8 -g levels/wf/5.ia8.png -f ia8
       > tools/n64graphics -i build/us_pc/levels/wmotr/0.rgba16 -g levels/wmotr/0.rgba16.png -f rgba16
       > gcc -c -O2  -I include -I build/us_pc -I build/us_pc/include -I src -I . -DRAPI_GL=1 -DWAPI_SDL2=1 -DAAPI_SDL2=1 -DCAPI_SDL2=1 -DHAVE_SDL2=1 -I/SDL2 -I/nix/store/vjrb1ayf85l4b7xr21fi9kkdgxrk4fnc-sdl2-compat-2.32.52-dev/include/SDL2 -I/nix/store/vjrb1ayf85l4b7xr21fi9kkdgxrk4fnc-sdl2-compat-2.32.52-dev/include/SDL2 -D_GNU_SOURCE=1 -D_REENTRANT -DVERSION_US -D_LANGUAGE_C -DNON_MATCHING -DAVOID_UB -DF3DEX_GBI_2E -fno-strict-aliasing -fwrapv -DEXT_OPTIONS_MENU -o build/us_pc/src/pc/platform.o src/pc/platform.c
       > tools/n64graphics -i build/us_pc/levels/wmotr/1.rgba16 -g levels/wmotr/1.rgba16.png -f rgba16
       > tools/n64graphics -i build/us_pc/levels/wmotr/2.rgba16 -g levels/wmotr/2.rgba16.png -f rgba16
       > tools/n64graphics -i build/us_pc/levels/wmotr/3.rgba16 -g levels/wmotr/3.rgba16.png -f rgba16
       > tools/n64graphics -i build/us_pc/levels/wmotr/4.rgba16 -g levels/wmotr/4.rgba16.png -f rgba16
       > make: *** [Makefile:1020: build/us_pc/src/pc/gfx/gfx_sdl2.o] Error 1
       > make: *** Waiting for unfinished jobs....
       > gcc -c -O2  -I include -I build/us_pc -I build/us_pc/include -I src -I . -DRAPI_GL=1 -DWAPI_SDL2=1 -DAAPI_SDL2=1 -DCAPI_SDL2=1 -DHAVE_SDL2=1 -I/SDL2 -I/nix/store/vjrb1ayf85l4b7xr21fi9kkdgxrk4fnc-sdl2-compat-2.32.52-dev/include/SDL2 -I/nix/store/vjrb1ayf85l4b7xr21fi9kkdgxrk4fnc-sdl2-compat-2.32.52-dev/include/SDL2 -D_GNU_SOURCE=1 -D_REENTRANT -DVERSION_US -D_LANGUAGE_C -DNON_MATCHING -DAVOID_UB -DF3DEX_GBI_2E -fno-strict-aliasing -fwrapv -DEXT_OPTIONS_MENU -o build/us_pc/levels/sa/leveldata.o levels/sa/leveldata.c
       > gcc -c -O2  -I include -I build/us_pc -I build/us_pc/include -I src -I . -DRAPI_GL=1 -DWAPI_SDL2=1 -DAAPI_SDL2=1 -DCAPI_SDL2=1 -DHAVE_SDL2=1 -I/SDL2 -I/nix/store/vjrb1ayf85l4b7xr21fi9kkdgxrk4fnc-sdl2-compat-2.32.52-dev/include/SDL2 -I/nix/store/vjrb1ayf85l4b7xr21fi9kkdgxrk4fnc-sdl2-compat-2.32.52-dev/include/SDL2 -D_GNU_SOURCE=1 -D_REENTRANT -DVERSION_US -D_LANGUAGE_C -DNON_MATCHING -DAVOID_UB -DF3DEX_GBI_2E -fno-strict-aliasing -fwrapv -DEXT_OPTIONS_MENU -o build/us_pc/src/pc/controller/controller_sdl2.o src/pc/controller/controller_sdl2.c
       > gcc -c -O2  -I include -I build/us_pc -I build/us_pc/include -I src -I . -DRAPI_GL=1 -DWAPI_SDL2=1 -DAAPI_SDL2=1 -DCAPI_SDL2=1 -DHAVE_SDL2=1 -I/SDL2 -I/nix/store/vjrb1ayf85l4b7xr21fi9kkdgxrk4fnc-sdl2-compat-2.32.52-dev/include/SDL2 -I/nix/store/vjrb1ayf85l4b7xr21fi9kkdgxrk4fnc-sdl2-compat-2.32.52-dev/include/SDL2 -D_GNU_SOURCE=1 -D_REENTRANT -DVERSION_US -D_LANGUAGE_C -DNON_MATCHING -DAVOID_UB -DF3DEX_GBI_2E -fno-strict-aliasing -fwrapv -DEXT_OPTIONS_MENU -o build/us_pc/levels/castle_inside/script.o levels/castle_inside/script.c
       > gcc -c -O2  -I include -I build/us_pc -I build/us_pc/include -I src -I . -DRAPI_GL=1 -DWAPI_SDL2=1 -DAAPI_SDL2=1 -DCAPI_SDL2=1 -DHAVE_SDL2=1 -I/SDL2 -I/nix/store/vjrb1ayf85l4b7xr21fi9kkdgxrk4fnc-sdl2-compat-2.32.52-dev/include/SDL2 -I/nix/store/vjrb1ayf85l4b7xr21fi9kkdgxrk4fnc-sdl2-compat-2.32.52-dev/include/SDL2 -D_GNU_SOURCE=1 -D_REENTRANT -DVERSION_US -D_LANGUAGE_C -DNON_MATCHING -DAVOID_UB -DF3DEX_GBI_2E -fno-strict-aliasing -fwrapv -DEXT_OPTIONS_MENU -o build/us_pc/levels/bits/script.o levels/bits/script.c
       > gcc -c -O2  -I include -I build/us_pc -I build/us_pc/include -I src -I . -DRAPI_GL=1 -DWAPI_SDL2=1 -DAAPI_SDL2=1 -DCAPI_SDL2=1 -DHAVE_SDL2=1 -I/SDL2 -I/nix/store/vjrb1ayf85l4b7xr21fi9kkdgxrk4fnc-sdl2-compat-2.32.52-dev/include/SDL2 -I/nix/store/vjrb1ayf85l4b7xr21fi9kkdgxrk4fnc-sdl2-compat-2.32.52-dev/include/SDL2 -D_GNU_SOURCE=1 -D_REENTRANT -DVERSION_US -D_LANGUAGE_C -DNON_MATCHING -DAVOID_UB -DF3DEX_GBI_2E -fno-strict-aliasing -fwrapv -DEXT_OPTIONS_MENU -o build/us_pc/levels/bowser_3/script.o levels/bowser_3/script.c
       > gcc -c -O2  -I include -I build/us_pc -I build/us_pc/include -I src -I . -DRAPI_GL=1 -DWAPI_SDL2=1 -DAAPI_SDL2=1 -DCAPI_SDL2=1 -DHAVE_SDL2=1 -I/SDL2 -I/nix/store/vjrb1ayf85l4b7xr21fi9kkdgxrk4fnc-sdl2-compat-2.32.52-dev/include/SDL2 -I/nix/store/vjrb1ayf85l4b7xr21fi9kkdgxrk4fnc-sdl2-compat-2.32.52-dev/include/SDL2 -D_GNU_SOURCE=1 -D_REENTRANT -DVERSION_US -D_LANGUAGE_C -DNON_MATCHING -DAVOID_UB -DF3DEX_GBI_2E -fno-strict-aliasing -fwrapv -DEXT_OPTIONS_MENU -o build/us_pc/levels/bowser_1/script.o levels/bowser_1/script.c
       > gcc -c -O2  -I include -I build/us_pc -I build/us_pc/include -I src -I . -DRAPI_GL=1 -DWAPI_SDL2=1 -DAAPI_SDL2=1 -DCAPI_SDL2=1 -DHAVE_SDL2=1 -I/SDL2 -I/nix/store/vjrb1ayf85l4b7xr21fi9kkdgxrk4fnc-sdl2-compat-2.32.52-dev/include/SDL2 -I/nix/store/vjrb1ayf85l4b7xr21fi9kkdgxrk4fnc-sdl2-compat-2.32.52-dev/include/SDL2 -D_GNU_SOURCE=1 -D_REENTRANT -DVERSION_US -D_LANGUAGE_C -DNON_MATCHING -DAVOID_UB -DF3DEX_GBI_2E -fno-strict-aliasing -fwrapv -DEXT_OPTIONS_MENU -o build/us_pc/levels/bbh/script.o levels/bbh/script.c
       > gcc -c -O2  -I include -I build/us_pc -I build/us_pc/include -I src -I . -DRAPI_GL=1 -DWAPI_SDL2=1 -DAAPI_SDL2=1 -DCAPI_SDL2=1 -DHAVE_SDL2=1 -I/SDL2 -I/nix/store/vjrb1ayf85l4b7xr21fi9kkdgxrk4fnc-sdl2-compat-2.32.52-dev/include/SDL2 -I/nix/store/vjrb1ayf85l4b7xr21fi9kkdgxrk4fnc-sdl2-compat-2.32.52-dev/include/SDL2 -D_GNU_SOURCE=1 -D_REENTRANT -DVERSION_US -D_LANGUAGE_C -DNON_MATCHING -DAVOID_UB -DF3DEX_GBI_2E -fno-strict-aliasing -fwrapv -DEXT_OPTIONS_MENU -o build/us_pc/levels/castle_courtyard/script.o levels/castle_courtyard/script.c
       > gcc -c -O2  -I include -I build/us_pc -I build/us_pc/include -I src -I . -DRAPI_GL=1 -DWAPI_SDL2=1 -DAAPI_SDL2=1 -DCAPI_SDL2=1 -DHAVE_SDL2=1 -I/SDL2 -I/nix/store/vjrb1ayf85l4b7xr21fi9kkdgxrk4fnc-sdl2-compat-2.32.52-dev/include/SDL2 -I/nix/store/vjrb1ayf85l4b7xr21fi9kkdgxrk4fnc-sdl2-compat-2.32.52-dev/include/SDL2 -D_GNU_SOURCE=1 -D_REENTRANT -DVERSION_US -D_LANGUAGE_C -DNON_MATCHING -DAVOID_UB -DF3DEX_GBI_2E -fno-strict-aliasing -fwrapv -DEXT_OPTIONS_MENU -o build/us_pc/src/pc/audio/audio_sdl2.o src/pc/audio/audio_sdl2.c
       > gcc -c -O2  -I include -I build/us_pc -I build/us_pc/include -I src -I . -DRAPI_GL=1 -DWAPI_SDL2=1 -DAAPI_SDL2=1 -DCAPI_SDL2=1 -DHAVE_SDL2=1 -I/SDL2 -I/nix/store/vjrb1ayf85l4b7xr21fi9kkdgxrk4fnc-sdl2-compat-2.32.52-dev/include/SDL2 -I/nix/store/vjrb1ayf85l4b7xr21fi9kkdgxrk4fnc-sdl2-compat-2.32.52-dev/include/SDL2 -D_GNU_SOURCE=1 -D_REENTRANT -DVERSION_US -D_LANGUAGE_C -DNON_MATCHING -DAVOID_UB -DF3DEX_GBI_2E -fno-strict-aliasing -fwrapv -DEXT_OPTIONS_MENU -o build/us_pc/levels/bitfs/script.o levels/bitfs/script.c
       > gcc -c -O2  -I include -I build/us_pc -I build/us_pc/include -I src -I . -DRAPI_GL=1 -DWAPI_SDL2=1 -DAAPI_SDL2=1 -DCAPI_SDL2=1 -DHAVE_SDL2=1 -I/SDL2 -I/nix/store/vjrb1ayf85l4b7xr21fi9kkdgxrk4fnc-sdl2-compat-2.32.52-dev/include/SDL2 -I/nix/store/vjrb1ayf85l4b7xr21fi9kkdgxrk4fnc-sdl2-compat-2.32.52-dev/include/SDL2 -D_GNU_SOURCE=1 -D_REENTRANT -DVERSION_US -D_LANGUAGE_C -DNON_MATCHING -DAVOID_UB -DF3DEX_GBI_2E -fno-strict-aliasing -fwrapv -DEXT_OPTIONS_MENU -o build/us_pc/levels/bowser_2/script.o levels/bowser_2/script.c
       > gcc -c -O2  -I include -I build/us_pc -I build/us_pc/include -I src -I . -DRAPI_GL=1 -DWAPI_SDL2=1 -DAAPI_SDL2=1 -DCAPI_SDL2=1 -DHAVE_SDL2=1 -I/SDL2 -I/nix/store/vjrb1ayf85l4b7xr21fi9kkdgxrk4fnc-sdl2-compat-2.32.52-dev/include/SDL2 -I/nix/store/vjrb1ayf85l4b7xr21fi9kkdgxrk4fnc-sdl2-compat-2.32.52-dev/include/SDL2 -D_GNU_SOURCE=1 -D_REENTRANT -DVERSION_US -D_LANGUAGE_C -DNON_MATCHING -DAVOID_UB -DF3DEX_GBI_2E -fno-strict-aliasing -fwrapv -DEXT_OPTIONS_MENU -o build/us_pc/levels/bitdw/script.o levels/bitdw/script.c
       > gcc -c -O2  -I include -I build/us_pc -I build/us_pc/include -I src -I . -DRAPI_GL=1 -DWAPI_SDL2=1 -DAAPI_SDL2=1 -DCAPI_SDL2=1 -DHAVE_SDL2=1 -I/SDL2 -I/nix/store/vjrb1ayf85l4b7xr21fi9kkdgxrk4fnc-sdl2-compat-2.32.52-dev/include/SDL2 -I/nix/store/vjrb1ayf85l4b7xr21fi9kkdgxrk4fnc-sdl2-compat-2.32.52-dev/include/SDL2 -D_GNU_SOURCE=1 -D_REENTRANT -DVERSION_US -D_LANGUAGE_C -DNON_MATCHING -DAVOID_UB -DF3DEX_GBI_2E -fno-strict-aliasing -fwrapv -DEXT_OPTIONS_MENU -o build/us_pc/levels/bob/script.o levels/bob/script.c
       > gcc -c -O2  -I include -I build/us_pc -I build/us_pc/include -I src -I . -DRAPI_GL=1 -DWAPI_SDL2=1 -DAAPI_SDL2=1 -DCAPI_SDL2=1 -DHAVE_SDL2=1 -I/SDL2 -I/nix/store/vjrb1ayf85l4b7xr21fi9kkdgxrk4fnc-sdl2-compat-2.32.52-dev/include/SDL2 -I/nix/store/vjrb1ayf85l4b7xr21fi9kkdgxrk4fnc-sdl2-compat-2.32.52-dev/include/SDL2 -D_GNU_SOURCE=1 -D_REENTRANT -DVERSION_US -D_LANGUAGE_C -DNON_MATCHING -DAVOID_UB -DF3DEX_GBI_2E -fno-strict-aliasing -fwrapv -DEXT_OPTIONS_MENU -o build/us_pc/levels/castle_grounds/script.o levels/castle_grounds/script.c
       > src/pc/platform.c: In function 'copy_userdata.constprop.isra':
       > src/pc/platform.c:102:34: warning: '/sm64_save_file.bin' directive output may be truncated writing 19 bytes into a region of size between 1 and 1024 [-Wformat-truncation=]
       >   102 |     snprintf(path, sizeof(path), "%s/" SAVE_FILENAME, userdir);
       >       |                                  ^~~~~
       > src/pc/platform.c:102:37: note: format string is defined here
       >   102 |     snprintf(path, sizeof(path), "%s/" SAVE_FILENAME, userdir);
       > In file included from /nix/store/pz6jgrfqyl2il2785a75vmwdp9vypv0j-glibc-2.40-66-dev/include/stdio.h:970,
       >                  from src/pc/platform.c:2:
       > In function 'snprintf',
       >     inlined from 'copy_userdata.constprop.isra' at src/pc/platform.c:102:5:
       > /nix/store/pz6jgrfqyl2il2785a75vmwdp9vypv0j-glibc-2.40-66-dev/include/bits/stdio2.h:68:10: note: '__builtin___snprintf_chk' output between 20 and 1043 bytes into a destination of size 1024
       >    68 |   return __builtin___snprintf_chk (__s, __n, __USE_FORTIFY_LEVEL - 1,
       >       |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
       >    69 |                                    __glibc_objsize (__s), __fmt,
       >       |                                    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
       >    70 |                                    __va_arg_pack ());
       >       |                                    ~~~~~~~~~~~~~~~~~
```

### Additional context

_No response_

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.14.2, NixOS, 25.05 (Warbler), 25.05.20250412.2631b0b`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.28.1`
 - nixpkgs: `/nix/store/5ds20jm3x2s4z7wn3581r6lc9ybmh45b-source`

### Notify maintainers

@qubitnano

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/sm/sm64ex/package.nix
+++ b/pkgs/by-name/sm/sm64ex/package.nix
@@ -18,13 +18,13 @@ in
 stdenv.mkDerivation (finalAttrs: {
 
   pname = "sm64ex";
-  version = "0-unstable-2024-07-04";
+  version = "0-unstable-2024-12-17";
 
   src = fetchFromGitHub {
     owner = "sm64pc";
     repo = "sm64ex";
-    rev = "20bb444562aa1dba79cf6adcb5da632ba580eec3";
-    hash = "sha256-nw+F0upTetLqib5r5QxmcOauSJccpTydV3soXz9CHLQ=";
+    rev = "d7ca2c04364a6dd0dac58b47151e04e26887e6f0";
+    hash = "sha256-n3ecY97UB/fdTZpy78CB4DxyHyjK+L6AAuNpvnmVoss=";
   };
 
   patches = lib.optionals _60fps [

--- a/pkgs/by-name/sm/sm64ex/package.nix
+++ b/pkgs/by-name/sm/sm64ex/package.nix
@@ -7,6 +7,7 @@
   pkg-config,
   audiofile,
   SDL2,
+  libGL,
   hexdump,
   sm64baserom,
   region ? "us",
@@ -44,6 +45,7 @@ stdenv.mkDerivation (finalAttrs: {
   buildInputs = [
     audiofile
     SDL2
+    libGL
   ];
 
   enableParallelBuilding = true;

